tools/src/prebuilds/ExpoModulesCorePrebuilder.ts

```diff
@@ -0,0 +1,175 @@
+} from './XcodeProject';
+import { Flavor, Framework, XcodebuildSettings } from './XcodeProject.types';
+
+// Generates working files out from expo-modules-core folder and prevents CocoaPods generates unnecessary files inside expo-modules-core folder
+const OUT_OF_TREE_WORKING_DIR = path.join(getExpoRepositoryRootDir(), 'prebuild-ExpoModulesCore');
+const MODULEMAP_FILE = 'ExpoModulesCore.modulemap';
+const UMBRELLA_HEADER = 'ExpoModulesCore-umbrella.h';
+
+export function isExpoModulesCore(pkg: Package) {
+  return pkg.packageName === 'expo-modules-core';
+}
+
+export async function generateXcodeProjectAsync(dir: string, spec: ProjectSpec): Promise<string> {
+  const workingDir = OUT_OF_TREE_WORKING_DIR;
+  await fs.ensureDir(workingDir);
+  logger.log(`   Prebuilding expo-modules-core from ${workingDir}`);
+
+  // Links to expo-modules-core source
+  if (typeof spec.targets?.['ExpoModulesCore'].sources?.[0].path === 'string') {
+    spec.targets['ExpoModulesCore'].sources[0].path = path.join(
+      getExpoRepositoryRootDir(),
+      'packages',
+      'expo-modules-core',
+      'ios'
+    );
+  }
+  // Links to generated header from prebuilder script
+  spec.targets?.['ExpoModulesCore']?.sources?.push({
+    path: '',
+    createIntermediateGroups: true,
+    name: 'ExpoModulesCore-umbrella',
+    includes: ['**/*.h'],
+  });
+
+  // Reset search header paths from base settings and leverages CocoaPods' setup
+  if (spec.settings?.base['HEADER_SEARCH_PATHS']) {
+    spec.settings.base['HEADER_SEARCH_PATHS'] = '$(inherited)';
+  }
+
+  if (spec.settings?.base) {
+    spec.settings.base['MODULEMAP_FILE'] = MODULEMAP_FILE;
+    spec.settings.base['BUILD_LIBRARY_FOR_DISTRIBUTION'] = 'YES';
+  }
+
+  await createModulemapAsync(workingDir);
+  await createGeneratedHeaderAsync(workingDir);
+
+  const result = await XcodeGen.generateXcodeProjectAsync(workingDir, spec);
+
+  logger.log('   Installing Pods');
+  await createPodfileAsync(workingDir);
+  await podInstallAsync(workingDir);
+
+  return result;
+}
+
+export async function buildFrameworkAsync(
+  xcodeProject: XcodeProject,
+  target: string,
+  flavor: Flavor,
+  options?: XcodebuildSettings
+): Promise<Framework> {
+  await xcodeProject.xcodebuildAsync(
+    [
+      'build',
+      '-workspace',
+      `${xcodeProject.name}.xcworkspace`,
+      '-scheme',
+      `${target}_iOS`,
+      '-configuration',
+      flavor.configuration,
+      '-sdk',
+      flavor.sdk,
+      ...spreadArgs('-arch', flavor.archs),
+      '-derivedDataPath',
+      SHARED_DERIVED_DATA_DIR,
+    ],
+    options
+  );
+
+  const frameworkPath = flavorToFrameworkPath(target, flavor);
+  const stat = await fs.lstat(path.join(frameworkPath, target));
+
+  // `_CodeSignature` is apparently generated only for simulator, afaik we don't need it.
+  await fs.remove(path.join(frameworkPath, '_CodeSignature'));
+
+  return {
+    target,
+    flavor,
+    frameworkPath,
+    binarySize: stat.size,
+  };
+}
```

Comment:
Is there any difference compared to `XcodeProject.buildFrameworkAsync`? I couldn't spot it 😅